cosmic-applet-battery/src/backend/mod.rs

```diff
@@ -0,0 +1,221 @@
+use cosmic::iced::{self, futures::SinkExt, subscription};
+use std::fmt::Debug;
+use std::hash::Hash;
+use tokio::sync::mpsc::UnboundedReceiver;
+use tokio::sync::mpsc::UnboundedSender;
+use zbus::Connection;
+use zbus::Result;
+
+use self::power_daemon::PowerDaemonProxy;
+use self::power_profiles::PowerProfilesProxy;
+
+mod power_daemon;
+mod power_profiles;
+
+#[derive(PartialEq, Eq, Copy, Clone, Debug, Default)]
+pub enum Power {
+    Battery,
+    #[default]
+    Balanced,
+    Performance,
+}
+
+#[derive(Debug)]
+pub enum Backend<'a> {
+    S76PowerDaemon(PowerDaemonProxy<'a>),
+    PowerProfilesDaemon(PowerProfilesProxy<'a>),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub enum BackendType {
+    #[default]
+    S76PowerDaemon,
+    PowerProfilesDaemon,
+}
+
+impl BackendType {
+    fn next(self) -> Self {
+        match self {
+            Self::S76PowerDaemon => Self::PowerProfilesDaemon,
+            Self::PowerProfilesDaemon => Self::S76PowerDaemon,
+        }
+    }
+}
+
+pub async fn get_power_profile(daemon: Backend<'_>) -> Result<Power> {
+    match daemon {
+        Backend::S76PowerDaemon(p) => {
+            let power = p.get_profile().await?;
+            match power.as_str() {
+                "Battery" => Ok(Power::Battery),
+                "Balanced" => Ok(Power::Balanced),
+                "Performance" => Ok(Power::Performance),
+                _ => panic!("Unknown power profile: {}", power),
+            }
+        }
+        Backend::PowerProfilesDaemon(ppd) => {
+            let power = ppd.active_profile().await?;
+            match power.as_str() {
+                "power-saver" => Ok(Power::Battery),
+                "balanced" => Ok(Power::Balanced),
+                "performance" => Ok(Power::Performance),
+                _ => panic!("Unknown power profile: {}", power),
+            }
+        }
+    }
+}
+
+pub async fn set_power_profile(daemon: Backend<'_>, power: Power) -> Result<()> {
+    println!("Changing power profile to {:?}", power);
```

Comment:
This should be removed or made a low level log statement like info or trace

Reply:
Fixed